src/main/java/ch/njol/skript/lang/ExpressionList.java

```diff
@@ -53,10 +54,19 @@ public ExpressionList(Expression<? extends T>[] expressions, Class<T> returnType
 		this(expressions, returnType, and, null);
 	}
 
+	public ExpressionList(Expression<? extends T>[] expressions, Class<T> returnType, Class<?>[] possibleReturnTypes, boolean and) {
```

Comment:
The incoming cast is functionally impossible because of the wildcard. You'd have to cast to object first and then probably use a type parameter to force it.
It also just doesn't provide any value because we still need to cast the individual things.